Accidental strict dep on LatencyUtils

It seems there are some guards to try to not use LatencyUtils unless pause detection is in use. However, there's still a compile dep.

```
[INFO] Caused by: java.lang.ClassNotFoundException: org.LatencyUtils.IntervalEstimator
[INFO] 	at java.net.URLClassLoader.findClass(URLClassLoader.java:471) ~[?:?]
[INFO] 	at java.lang.ClassLoader.loadClass(ClassLoader.java:588) ~[?:?]
[INFO] 	at org.springframework.boot.loader.LaunchedURLClassLoader.loadClass(LaunchedURLClassLoader.java:93) ~[zipkin-server-2.16.3-SNAPSHOT-slim.jar:?]
[INFO] 	at java.lang.ClassLoader.loadClass(ClassLoader.java:521) ~[?:?]
[INFO] 	at io.micrometer.prometheus.PrometheusMeterRegistry.newTimer(PrometheusMeterRegistry.java:176) ~[micrometer-registry-prometheus-1.2.1.jar!/:1.2.1]
[INFO] 	at io.micrometer.core.instrument.MeterRegistry.lambda$timer$2(MeterRegistry.java:271) ~[micrometer-core-1.2.1.jar!/:1.2.1]
[INFO] 	at io.micrometer.core.instrument.MeterRegistry.getOrCreateMeter(MeterRegistry.java:576) ~[micrometer-core-1.2.1.jar!/:1.2.1]
[INFO] 	at io.micrometer.core.instrument.MeterRegistry.registerMeterIfNecessary(MeterRegistry.java:529) ~[micrometer-core-1.2.1.jar!/:1.2.1]
[INFO] 	at io.micrometer.core.instrument.MeterRegistry.timer(MeterRegistry.java:269) ~[micrometer-core-1.2.1.jar!/:1.2.1]
[INFO] 	at io.micrometer.core.instrument.Timer$Builder.register(Timer.java:477) ~[micrometer-core-1.2.1.jar!/:1.2.1]
[INFO] 	at io.micrometer.core.instrument.MeterRegistry.timer(MeterRegistry.java:396) ~[micrometer-core-1.2.1.jar!/:1.2.1]
[INFO] 	at io.micrometer.core.instrument.internal.TimedExecutorService.<init>(TimedExecutorService.java:40) ~[micrometer-core-1.2.1.jar!/:1.2.1]
[INFO] 	at io.micrometer.core.instrument.binder.jvm.ExecutorServiceMetrics.monitor(ExecutorServiceMetrics.java:94) ~[micrometer-core-1.2.1.jar!/:1.2.1]
[INFO] 	at io.micrometer.core.instrument.binder.jvm.ExecutorServiceMetrics.monitor(ExecutorServiceMetrics.java:107) ~[micrometer-core-1.2.1.jar!/:1.2.1]
```

--- a/micrometer-core/src/main/java/io/micrometer/core/instrument/AbstractTimer.java
+++ b/micrometer-core/src/main/java/io/micrometer/core/instrument/AbstractTimer.java
@@ -17,6 +17,7 @@ package io.micrometer.core.instrument;
 
 import io.micrometer.core.instrument.distribution.*;
 import io.micrometer.core.instrument.distribution.pause.ClockDriftPauseDetector;
+import io.micrometer.core.instrument.distribution.pause.NoPauseDetector;
 import io.micrometer.core.instrument.distribution.pause.PauseDetector;
 import io.micrometer.core.lang.Nullable;
 import org.LatencyUtils.IntervalEstimator;
@@ -101,6 +102,9 @@ public abstract class AbstractTimer extends AbstractMeter implements Timer {
     }
 
     private void initPauseDetector(PauseDetector pauseDetectorType) {
+        if (pauseDetectorType instanceof NoPauseDetector) {
+            return;
+        }
         pauseDetector = pauseDetectorCache.computeIfAbsent(pauseDetectorType, detector -> {
             if (detector instanceof ClockDriftPauseDetector) {
                 ClockDriftPauseDetector clockDriftPauseDetector = (ClockDriftPauseDetector) detector;

--- a/samples/micrometer-samples-boot2/build.gradle
+++ b/samples/micrometer-samples-boot2/build.gradle
@@ -10,7 +10,11 @@ repositories {
 }
 
 dependencies {
-    implementation project(":micrometer-core")
+    implementation(project(":micrometer-core")) {
+        // see gh-1599; pause detection and percentiles are not configured so these dependencies can be excluded
+        exclude module: 'LatencyUtils'
+        exclude module: 'HdrHistogram'
+    }
     ['atlas', 'azure-monitor', 'prometheus', 'datadog', 'elastic', 'ganglia', 'graphite', 'health', 'jmx', 'influx', 'statsd', 'new-relic', 'cloudwatch', 'cloudwatch2', 'signalfx', 'wavefront', 'elastic', 'dynatrace', 'humio', 'appoptics', 'stackdriver'].each { sys ->
         implementation project(":micrometer-registry-$sys")
     }

